### DISCUSSION TITLE
How to get data from the triggering item

### DISCUSSION THREAD

**Discussion by craigpotter** [fancyapps/ui#21] [Q&A] [answered]
Hi, 

Loving the new version! 
I've been looking at the show case and at the [two panel layout example](https://fancyapps.com/playground/gj).
.
If you were to add addition data to the click item, an ID or something to then get the content dynamically how would you access the triggering item. 

For example

```html 

<div class="flex flex-wrap gap-5 justify-center max-w-5xl mx-auto px-6">
  <a data-fancybox="gallery" data-id="5" href="https://lipsum.app/id/63/800x600">
    <img class="rounded" src="https://lipsum.app/id/63/200x150" />
  </a>
  <a data-fancybox="gallery" data-id="6" href="https://lipsum.app/id/64/800x600">
    <img class="rounded" src="https://lipsum.app/id/64/200x150" />
  </a>
</div>
```

In the `initLayout` event, you can access the Fancybox instance by passing it in but I am unsure how to access the trigging element to get the `data-id`. 

Am I missing something in the docs ?

**Comment by fancyapps** [comment, marked as answer]
Hi,

It would be like this:

```JS
Fancybox.bind('[data-fancybox="gallery"]', {
  on : {
    initLayout : (fancybox) => {
      console.log(fancybox.options.$trigger);
    }
  }
});
```

Sorry, the docs are not complete yet. I am also planning to create a tutorial that would cover frequently asked questions.

**Reply by craigpotter** [reply]
Thanks for the fast reply! 

I swear I did `console.log(fancybox)` and looked through everything ... not sure how I missed that!
